packages/block-library/src/image/save.js

```diff
@@ -49,20 +49,26 @@ export default function save( { attributes } ) {
 		[ `wp-image-${ id }` ]: !! id,
 	} );
 
+	const imgStyle = {
+		...borderProps.style,
+		aspectRatio,
+		objectFit: scale,
+	};
+	if ( width ) {
+		imgStyle.width = width;
+	}
+
+	if ( height && ! isNaN( height ) ) {
+		imgStyle.height = height;
+	}
 	const image = (
 		<img
 			src={ url }
 			alt={ alt }
 			className={ imageClasses || undefined }
-			style={ {
-				...borderProps.style,
-				aspectRatio,
-				objectFit: scale,
-				width,
-				height,
-			} }
+			style={ imgStyle }
 			width={ width }
-			height={ height }
+			height={ isNaN( height ) ? undefined : height }
```

Comment:
In order to prevent layout shifts when `height` is `'auto'`, we need to ensure that this property contains a real value.

Reply:
We could keep the height for the image attributes and just use auto in the CSS